### PR TITLE
FIX many sql injections

### PR DESCRIPTION
**Pull request by all3kcis**
[Really dangerous - Important]
It was also possible to set any user "admin", just with user right "user->rights->user->self->creer".

[Important]
With just the user right "user-> create" it was possible to create a user with login and password empty and connect with this.

(int) use to type int, and tinyint in DB,
faster than db-> escape and sufficient.

Existing in Dolibarr develop, 3.7.x, 3.6.x not tested on older version.

I did check that classes present here.
I don't think to have all seen and corrected.
I encourage to check my changes

### CODE REVIEW

File: htdocs/commande/card.php [hunk 1]
```diff
@@ -241,19 +241,19 @@
...
 			$object->note_public = GETPOST('note_public');
-			$object->source = GETPOST('source_id');
-			$object->fk_project = GETPOST('projectid');
-			$object->ref_client = GETPOST('ref_client');
-			$object->modelpdf = GETPOST('model');
-			$object->cond_reglement_id = GETPOST('cond_reglement_id');
-			$object->mode_reglement_id = GETPOST('mode_reglement_id');
+			$object->source = GETPOST('source_id', 'int');
+			$object->fk_project = GETPOST('projectid', 'int');
+			$object->ref_client = GETPOST('ref_client', 'int');
```
Comment (hregis) [inline htdocs/commande/card.php]: ref_client is not an integer

Comment (all3kcis) [inline htdocs/commande/card.php]: exact thank you

**Comment by all3kcis** [comment]
Update first message :
"[Really dangerous - Important]
It was also possible to set any user "admin", just with user right "user->rights->user->self->creer"."

**Comment by rdoursenaud** [comment]
Since this is obviously very important and we are in a beta cycle for 3.8, would you mind rebasing your changes over to a 3.8 branch and make a pull request to 3.8.

Thanks!

**Comment by all3kcis** [comment]
Yes ok

**Comment by all3kcis** [comment]
Thank you wait to merge, more changes coming.

**Comment by all3kcis** [comment]
I know few phpunit, but I think the problem can be set with #3303

**Comment by eldy** [comment]
Ldap_sidhas been set to int. I think it may be hex so with chars.

**Comment by eldy** [comment]
Also i can see fix to creation of admin user. It is a diferent bug, can you make a different pb for it ?

**Comment by all3kcis** [comment]
You want me to create a new pull request just for the admin injection ?
I therefore consider "Ldap_sid" as int ?

**Comment by hregis** [comment]
@All-3kcis @eldy 
ldap sid example : 1-5-21-3528217606-2650276867-2060070683-1111

**Comment by all3kcis** [comment]
See #3327 , just for the correction of "admin injection"
I'm on vacation, I continued the corrections to my return.

**Comment by all3kcis** [comment]
Can you tell me more about the problem with travis test?

**Comment by eldy** [comment]
The PR seems to introduce regressions.
For example into test WebservicesUserTest::testWSUserGetUser. You can click on link "Details" and see at end of report to have details of errors

**Comment by eldy** [comment]
Due to conflict and regressions, i manually introduced fixes i thought it was safe into another commit.

**Comment by FHenry** [comment]
variable casting ! Cool. We also should also update php_build_my_class.php to include this new code usage

**Pull request closed without merging**